### DISCUSSION TITLE
Subscribe to `CollectionChanged` event for `IList` properties

### DISCUSSION THREAD

**Discussion by AndreasReitberger** [realm/realm-dotnet#3257] [Q&A]
I have a simple object which has a `IList` as property.
In this class, I want to subscribe to the `CollectionChanged` event, however I haven't found a proper way to do so.

```cs
public partial class MyObject : RealmObject
{
  public IList<SecondObject> MyList { get; }

  public MyObject() {
       MyList .AsRealmCollection().CollectionChanged += MyList_CollectionChanged;
  }

 void MyList_CollectionChanged(object? sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
        {
            // Do stuff
        }
}
```
If I do it this way, I get an exception in the ctor of `MyObject`.

> System.ArgumentException: "list must be an instance of IRealmCollection<Stock>. Arg_ParamName_Name"

**Comment by papafe** [comment]
Hi @AndreasReitberger, the problem here is that you're trying to subscribe for notifications when the object is unmanaged (has not been added to a realm). If you do it after you've added your instance of `MyObject` to a realm, then it will work.

Just to give you an idea of why it's happening, when the object gets managed, we "swap" out the list implementation for a list that implements `IRealmCollection`.

**Reply by AndreasReitberger** [reply]
OK I see and understand. Is there a way to subscribe inside the `MyObject` to changes? In my old code, I just used  a `ObservableCollection`.
Is there any event I can use which gets called if the "swap" was done?
I need to update some other properties if the list property changes, so the `MyObject` should automatically subscribe the event once it's created.

**Reply by papafe** [reply]
If you need that you can override `OnManaged()`, that's called when the object gets managed.

**Reply by nirinchev** [reply]
Keep in mind that you may end up with multiple objects attempting to do the same thing - e.g. if you have multiple instances of the object, they'll all attempt to handle the same change event.
